users/utils.py

```diff
@@ -37,3 +42,47 @@ def user_has_change_permission(user, *models):
 
 def user_has_delete_permission(user, *models):
     return _user_has_models_perms("delete", models)(user)
+
+
+def is_customer(user):
+    if user.is_staff or user.is_superuser:
+        return False
+    elif (
+        user.groups.count() == 1
+        and user.groups.first().name == settings.CUSTOMER_GROUP_NAME
+    ):
+        return True
+    else:
+        return False
+
+
+def get_berth_customers_group():
+    return Group.objects.get(name=settings.CUSTOMER_GROUP_NAME)
+
+
+def get_node_user(node):
+    """
+       Try to get to the user this node belongs to
+       either directly by node.user or through CustomerProfile,
+       i.e. by doing node.customer.user
+    """
+    if not node:
+        return None
+
+    node_user = None
+    if hasattr(node, "user"):
+        node_user = node.user
+    elif hasattr(node, "customer") and hasattr(node.customer, "user"):
+        node_user = node.customer.user
+    return node_user
+
+
+def user_is_linked_to_node(user, node):
+    if user is None:
+        return False
+    return user == get_node_user(node)
+
+
+def validate_user_is_linked_to_node(user, node):
+    if not user_is_linked_to_node(user, node):
+        raise PermissionDenied(_("You do not have permission to perform this action."))
```

Comment:
Where are we gonna use this?

Reply:
It would be useful for mutations where customer wants to edit an object. For example, in VEN-1222 it would be used to check if the current user is the owner of the application that's being edited.